### PR TITLE
ch122170 Download the latest c-server-sdk release instead of a fixed version.

### PR DESCRIPTION
**Pull request by kinyoklion**
This PR updates the `download.sh` script to download the latest version of the `c-server-sdk`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,14 +37,14 @@ jobs:
...
           name: Download SDK
           command: ./download.sh
       - run:
           name: Build SDK
-          command: luarocks make launchdarkly-server-sdk-1.0-0.rockspec LD_DIR=. LD_INCDIR=./include
+          command: luarocks --lua-version=5.1 make launchdarkly-server-sdk-1.0-0.rockspec LD_DIR=. LD_INCDIR=./include
       - run:
           name: Run hello
           command: |
-            LD_LIBRARY_PATH=. lua5.3 hello.lua
```
Comment (kinyoklion) [inline .circleci/config.yml:50]: Nothing was installing lua, or specifically 5.3, so this failed. I used 5.1 for each step for now.

Comment (kinyoklion) [reply .circleci/config.yml:50]: Linux was fine, because usbuntu 18.04 will default to a compatible version of lua.

File: download.sh [hunk 2]
```diff
@@ -4,12 +4,20 @@ set -e
 if [[ "$OSTYPE" == "linux-gnu" ]]; then
     echo 'downloading linux c-server-sdk shared library'
     rm -rf 'linux-gcc-64bit-dynamic.tar' 'include' 'lib'
-    curl -s -o 'linux-gcc-64bit-dynamic.tar' -L 'https://github.com/launchdarkly/c-server-sdk/releases/download/2.2.1/linux-gcc-64bit-dynamic.tar'
+    curl https://api.github.com/repos/launchdarkly/c-server-sdk/releases/latest | \
```
Comment (kinyoklion) [inline download.sh:7]: Use the github to get the latest release. Then from that release find the linux or mac donwload URL. Then pipe that URL back into curl and download it.

File: download.sh [hunk 3]
```diff
@@ -4,12 +4,20 @@ set -e
...
 elif [[ "$OSTYPE" == "darwin"* ]]; then
     echo 'downloading osx c-server-sdk shared library'
     rm -rf 'osx-clang-64bit-dynamic.tar' 'include' 'lib'
-    curl -s -o "osx-clang-64bit-dynamic.tar" -L "https://github.com/launchdarkly/c-server-sdk/releases/download/2.2.1/osx-clang-64bit-dynamic.tar"
+    curl https://api.github.com/repos/launchdarkly/c-server-sdk/releases/latest | \
+    grep browser_download_url | \
+    grep osx-clang-64bit-dynamic.tar | \
+    cut -d '"' -f 4 | \
+    xargs curl -s -o 'osx-clang-64bit-dynamic.tar' -L
     tar -xvf 'osx-clang-64bit-dynamic.tar'
```
Comment (bwoskow-ld) [inline download.sh:21]: You could refactor this into a bash function which takes a single filename parameter (`linux-gcc-64bit-dynamic.tar` | `osx-clang-64bit-dynamic.tar`). Up to you if you want to do so.

**Review by bwoskow-ld — APPROVED** [review]
Approved, but with some optional feedback which you can optionally choose to address.

**Review by gwhelanLD — COMMENTED** [review]
Did we discuss whether we should limit the usage of the latest release to a specific major version of the C SDK?

**Comment by kinyoklion** [comment]
> Did we discuss whether we should limit the usage of the latest release to a specific major version of the C SDK?

I do not think we did. How currently is the lua-server-sdk updated in relation to the c-server-sdk? The documentation of supported versions in the readme doesn't seem correct, not for the master of the branch at least.

**Review by gwhelanLD — APPROVED** [review]

**Pull request merged**